kquery/KQuery.g4

```diff
@@ -0,0 +1,357 @@
+    : varName                                                                               #VariableName
+    | namedConstant Colon expression                                                        #NamedAbbreviation
+    | LeftParen widthOrSizeExpr number RightParen                                           #SizeQuery
+    | LeftParen arithmeticExpr widthOrSizeExpr leftExpr rightExpr RightParen                #ArithExpr
+    | LeftParen notExpr LeftBracket widthOrSizeExpr RightBracket expression RightParen      #NotExprWidth
+    | LeftParen bitwiseExpr widthOrSizeExpr leftExpr rightExpr RightParen                   #BitwExprWidth
+    | LeftParen comparisonExpr widthOrSizeExpr leftExpr rightExpr RightParen                #CompExprWidth
+    | LeftParen comparisonExpr leftExpr rightExpr RightParen                                #CompExpr
+    | LeftParen concatExpr widthOrSizeExpr leftExpr rightExpr RightParen                    #ConcatExprWidth
+    | LeftParen concatExpr leftExpr rightExpr RightParen                                    #ConcatExprNW   
+    | LeftParen arrExtractExpr widthOrSizeExpr number expression RightParen                 #ArrExtractExprWidth
+    | LeftParen bitExtractExpr widthOrSizeExpr expression RightParen                        #BitExtractExprWidth
+    | LeftParen genericBitRead widthOrSizeExpr expression version RightParen                #ReadExpresssionVersioned
+    | LeftParen selectExpr widthOrSizeExpr leftExpr rightExpr expression RightParen         #SelectExprWidth
+    | LeftParen exprNegation widthOrSizeExpr expression RightParen                          #NegationExprWidth
+    | LeftParen exprNegation expression RightParen                                          #NegetionExpr
+    | LeftParen genericBitRead widthOrSizeExpr expression RightParen                        #ReadExpr
+    | version                                                                               #VersionExpr
+    | number                                                                                #Singleton
+    ;
+
+genericBitRead
+    : READ          
+    | READLSB
+    | READMSB
+    ;
+
+bitExtractExpr
+    : ZEXT
+    | SEXT
+    ;
+    
+version
+    : varName                                               #VersionVariableName
+    | namedConstant Colon expression                        #VersionedNamedAbbreviation
+    | LeftBracket updateList RightBracket ATR version       #UpdationList
+    | LeftBracket RightBracket ATR version                  #NoUpdateList
+    ;
+    
+notExpr
+    : NOT
+    ;
+    
+concatExpr
+    : CONCAT
+    ;
+
+exprNegation
+    : NEGETION
+    ;
+
+selectExpr
+    : SELECT
+    ;
+
+arrExtractExpr
+    : EXTRACT
+    ;
+    
+varName
+    : Identifier
+    ;
+    
+leftExpr 
+    : expression
+    ; 
+
+rightExpr
+    : expression
+    ;
+    
+namedConstant
+    : Identifier
+    ;
+    
+updateList 
+    : expression Equal expression COMMA updateList
+    | expression Equal expression
```

Comment:
`updateList : expression Equal expression ( COMMA expression Equal expression )* ;`

Reply:
There's likely other problems. The issue is that this grammar is awfully slow. That is why it fails for the PHP target--which is terribly slow to begin with. Antlr really is much faster if you use EBNF.

Reply:
BTW, the alternative is to add this to the [skip list for PHP](https://github.com/lahiri-phdworks/grammars-v4/blob/master/_scripts/skip-php.txt) and just say in the readme.md that it's "unoptimized". People will know what you mean.